src/jit/lsrabuild.cpp

```diff
@@ -25,86 +25,127 @@ XXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX
+// RefInfoList
+//------------------------------------------------------------------------
+// removeListNode - retrieve the RefInfoListNode for the given GenTree node
+//
+// Notes:
+//     The BuildNode methods use this helper to retrieve the TreeNodeInfo for child nodes
+//     from the useList being constructed. Note that, if the user knows the order of the operands,
+//     it is expected that they should just retrieve them directly.
+
+RefInfoListNode* RefInfoList::removeListNode(GenTree* node)
+{
+    RefInfoListNode* prevListNode = nullptr;
+    for (RefInfoListNode *listNode = Begin(), *end = End(); listNode != end; listNode = listNode->Next())
+    {
+        if (listNode->treeNode == node)
+        {
+            assert(listNode->ref->getMultiRegIdx() == 0);
+            return removeListNode(listNode, prevListNode);
+        }
+        prevListNode = listNode;
+    }
+    assert(!"GetRefPosition didn't find the node");
+    unreached();
+}
+
+//------------------------------------------------------------------------
+// removeListNode - retrieve the RefInfoListNode for one reg of the given multireg GenTree node
+//
+// Notes:
+//     The BuildNode methods use this helper to retrieve the TreeNodeInfo for child nodes
+//     from the useList being constructed. Note that, if the user knows the order of the operands,
+//     it is expected that they should just retrieve them directly.
+
+RefInfoListNode* RefInfoList::removeListNode(GenTree* node, unsigned multiRegIdx)
+{
+    RefInfoListNode* prevListNode = nullptr;
+    for (RefInfoListNode *listNode = Begin(), *end = End(); listNode != end; listNode = listNode->Next())
+    {
+        if ((listNode->treeNode == node) && (listNode->ref->getMultiRegIdx() == multiRegIdx))
+        {
+            return removeListNode(listNode, prevListNode);
+        }
+        prevListNode = listNode;
+    }
+    assert(!"GetRefPosition didn't find the node");
+    unreached();
+}
+
+//------------------------------------------------------------------------
+// RefInfoListNodePool::RefInfoListNodePool:
+//    Creates a pool of `RefInfoListNode` values.
 //
 // Arguments:
 //    compiler    - The compiler context.
 //    preallocate - The number of nodes to preallocate.
 //
-LocationInfoListNodePool::LocationInfoListNodePool(Compiler* compiler, unsigned preallocate) : m_compiler(compiler)
+RefInfoListNodePool::RefInfoListNodePool(Compiler* compiler, unsigned preallocate) : m_compiler(compiler)
 {
     if (preallocate > 0)
     {
-        size_t                preallocateSize = sizeof(LocationInfoListNode) * preallocate;
-        LocationInfoListNode* preallocatedNodes =
-            reinterpret_cast<LocationInfoListNode*>(compiler->compGetMem(preallocateSize, CMK_LSRA));
+        size_t           preallocateSize = sizeof(RefInfoListNode) * preallocate;
+        RefInfoListNode* preallocatedNodes =
+            reinterpret_cast<RefInfoListNode*>(compiler->compGetMem(preallocateSize, CMK_LSRA));
```

Comment:
Existing code but note that casting from `void*` to `X*` is normally done via `static_cast`. `reinterpret_cast` is usually reserved for "weird" stuff such as casting `int*` to `float*`.

Reply:
Made the change